security/pkg/server/ca/server_test.go

```diff
@@ -212,58 +212,76 @@ func TestHandleCSR(t *testing.T) {
 	testCases := map[string]struct {
 		authenticators []authenticator
 		authorizer     *mockAuthorizer
-		ca             ca.CertificateAuthority
+		ca             *mockca.FakeCA
```

Comment:
(Optional since it's not the focus of this PR) how hard to use real CA implementation here? Is the reason mostly due to APIServer secrets dependency?

Reply:
Yes for the dependencies. And also we want to make this unit test self-contained.

Reply:
I'm the fan of using real implementation and mock at low level dependencies. Never mind, this is very debatable :)

Reply:
Sure. We could improve it in the future :)